packages/core/src/interpreter.ts

```diff
@@ -718,9 +718,7 @@ export class Interpreter<
 
     if (
       _event.name.indexOf(actionTypes.errorPlatform) === 0 &&
-      !this.state.nextEvents.some(
-        (nextEvent) => nextEvent.indexOf(actionTypes.errorPlatform) === 0
-      )
+      !this.state.can(_event as any)
```

Comment:
Note that both the current logic (the one on the main branch) and the proposed logic in this PR sound incorrect to me, regardless of the decisions that have to be made in response to [the other comment](https://github.com/statelyai/xstate/pull/2935/files#r781981212).

Incorrect scenarios are really edge-case-y though and applicable more in v5 context than in v4. This whole if statement might be also dropped entirely as part of the work resulting from [The Errors RFC](https://github.com/statelyai/rfcs/pull/4) so it's not worth addressing this now.

With the current logic, based on `nextEvents`, this won't quite work correctly when we consider parallel regions because a single `onError` in one of the regions "turns off" this if statement for another parallel region.

The proposed logic feels incorrect because we can create such structure:
```js
on: {
  [errorDescriptor]: { /* do smth */ },
},
states: {
  a: {
    invoke: {
      src: 'fetch',
      onError: undefined
    }
  }
}
```
As we might see here - the inner error handler is forbidden but the outer one is not (the latter doesn't quite matter but maybe other edge cases can be created using such setup). This would return false from `state.can(errorEvent)` when in the `a` state but the `onError: undefined` looks explicit enough to count this as an "explicit" error handler. It feels that it would make sense to swallow errors with such a forbidden `onError` - note that it's not obvious how to express "error swallowing" differently. The only thing that comes to my mind is this:
```js
onError: { actions: noop }
```
And I wouldn't count this as intuitive.

Reply:
I added v5 essential label to this PR just to remember to incorporate thoughts from this comment above. The PR itself has to be redone.